In [1]:
# tree dataset
import pandas as pd
import statsmodels.api as sm
tree = sm.datasets.get_rdataset('trees').data
tree.head()

Girth  Height  Volume
0    8.3      70    10.3
1    8.6      65    10.3
2    8.8      63    10.2
3   10.5      72    16.4
4   10.7      81    18.8

In [2]:
tree.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 31 entries, 0 to 30
Data columns (total 3 columns):
 #   Column  Non-Null Count  Dtype  
---  ------  --------------  -----  
 0   Girth   31 non-null     float64
 1   Height  31 non-null     int64  
 2   Volume  31 non-null     float64
dtypes: float64(2), int64(1)
memory usage: 876.0 bytes


In [5]:
# creat a linear regression model with ols
model = sm.formula.ols(formula='Volume ~ Girth + Height', data=tree).fit()
print(model.summary())

                            OLS Regression Results                            
Dep. Variable:                 Volume   R-squared:                       0.948
Model:                            OLS   Adj. R-squared:                  0.944
Method:                 Least Squares   F-statistic:                     255.0
Date:                Mon, 25 Nov 2024   Prob (F-statistic):           1.07e-18
Time:                        16:43:18   Log-Likelihood:                -84.455
No. Observations:                  31   AIC:                             174.9
Df Residuals:                      28   BIC:                             179.2
Df Model:                           2                                         
Covariance Type:            nonrobust                                         
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
Intercept    -57.9877      8.638     -6.713      0.0

Note that our linear regression model reads "Volume = 4.7082*Girth + 0.3393*Height - 57.9877".
All the coefficients are statistically signifant as their corresponding p-values are lower then 0.05.
All these results are evaluated with R-sqaured = 0.948.

In [7]:
# with the use of the regression model, predic a volume of some trees whose girth and weight are known.
new_data = pd.DataFrame({'Girth':[8.5, 13.0, 19.0], 'Height':[72,86,85]})
vol_predict = model.predict(new_data)
vol_predict

0     6.457794
1    32.394034
2    60.303746
dtype: float64